### ISSUE TITLE
Login form doesn't show or be accessible, even with secret key

### ISSUE THREAD

**Issue by filipe-torres** [alve89/j_plg_hrz_disablelogin#8]
Hi Stefan,
I installed this plugin on **Joomla 5** website, but I can't access login form.

I follow plugin instructions:
1. Enable plugin
2. Set a Secret Key: safelogin
3. Save plugin settings
4. Unpublish hidden login menu item

Then I tried to access frontend login form by the following addresses, but no success:
https://creativante.com.br/?safelogin
https://creativante.com.br/?option=com_users&safelogin
https://creativante.com.br/index.php/component/users?safelogin

Where did I go wrong? I miss something?

**Comment by alve89** [comment]
Hi Filipe,

thank you for reaching out. Let's figure this out.

You write that
- the login form doesn't show
- or is accessible.

This raises the following questions:
1. Where is your login form located? I guess it's `/` (so `https://creativante.com.br/`), right?
2. Which user groups are permitted to access (= view) the module?

First solution: Try `/?option=com_users&view=login`.

**Comment by filipe-torres** [comment]
Hi Stephan,
Answering you questions: 
> 1. Where is your login form located? I guess it's `/` (so `https://creativante.com.br/`), right?
> 2. Which user groups are permitted to access (= view) the module?

1. Login form was located at https://creativante.com.br/login. I followed the instructions and unpublished this login menu item.
2. Login form has public access, otherwise unlogged users couldn't access to edit articles on frontend.

> First solution: Try `/?option=com_users&view=login`.

I got access to login form: https://creativante.com.br/?option=com_users&view=login
The problem is: this URL shouldn't be accessible, right? The main purpose of plugin is avoid access to login form using URL parameters **without secret key**. This URL and user component parameters should only work when used combined with secret key, "safelogin" on my specific case: https://creativante.com.br/?option=com_users&view=login&safelogin

Follow below plugin configuration screen:
![Captura de tela 2024-05-22 102950](https://github.com/alve89/j_plg_hrz_disablelogin/assets/1202645/709afb87-880d-4bf1-a997-1a4e8f452e9a)

**Comment by alve89** [comment]
> The problem is: this URL shouldn't be accessible, right?

Right. I assume you tested all URLs within the same session, so without closing the browser in between. Is this correct?

The way the plugin works is as follows (see [code](https://github.com/alve89/j_plg_hrz_disablelogin/blob/efd1b37209f79347e93796b2d7b75f73453aa033/disablelogin.php#L37C1-L44C8)):
1. Check if a `secretKey` is provided or else 
2. if a **correct** `secretKey` was provided once (in the same session).
3. If one of both options are true, the plugin stops further checks.

So if you want to run proper tests, close the browser completely after using the `secretKey` once, or use another browser parallel.

Further I'm still interested in your setup. You write that
> Login form was located at https://creativante.com.br/login. I followed the instructions and unpublished this login menu item.

So just to avoid misunderstandings: You don't have any login forms anymore (except of the system ones from the plugin description)? Then the plugin should work properly, except of one change from J4 to J5 (I'll add this information in the description after writing this comment):

While in J4 you were able to use `https://myDomain.tld/my/path/to/joomla?option=com_users&YOUR_SECRET_KEY` to login, in J5 you need to use
- `https://myDomain.tld/my/path/to/joomla?option=com_users&view=login&YOUR_SECRET_KEY` or
- `https://myDomain.tld/my/path/to/joomla/index.php/component/users?view=login&YOUR_SECRET_KEY`.

Otherwise a `404` system error is thrown because no view was specified:
![image](https://github.com/alve89/j_plg_hrz_disablelogin/assets/11487054/41e60600-b812-4edf-8502-0283a931f685)

Adding the `view=login` parameter, the error message is solved:
<img width="500" alt="image" src="https://github.com/alve89/j_plg_hrz_disablelogin/assets/11487054/d32cc9d2-dfa7-431b-87c4-84666c4233d5">

Let me know if any of these points are helpful for you to resolve your problem. Otherwise I'm glad to provide further help.

**Comment by filipe-torres** [comment]
Hi Stephan,
I followed your instructions and works like a charm!!!

Asking your question:
> So just to avoid misunderstandings: You don't have any login forms anymore (except of the system ones from the plugin description)?

Yes.

P.S.: Just an idea to possible/future plugin: after to use, install and test this plugin I realized that could be useful too if there is a similar plugin to restrict access to one or more URLs. Plugin would work very similar and block access to https://myDomain.tld/path but allow access if provide a correct secret key: https://myDomain.tld/path&YOUR_SECRET_KEY.
On plugin configuration form user should fill and set:

- URL (text input)
- Secret Key (text input)
- Add more (button) <-- to add a new URL and Secret Key
------------------------------------------------------------------
- Enable Logging (boolean)
- Show access denial in frontend (boolean)
- Target-address on blocked access (text input)

Thanks for your patience and disposal to help.
